task3/UI/script_task7.js

```diff
@@ -0,0 +1,527 @@
+class AdCollection {
+    constructor(adList) {
+        if (adList) {
+            this._adList = adList.slice();
+        } else {
+            this._adList = [];
+        }
+    }
+
+    add(adItem) {
+        if (AdCollection._validate(adItem) && !this.get(adItem.id)) {
+            this._adList.push(adItem);
+            return true;
+        }
+        return false;
+    }
+
+    addAll(adList) {
+        let notValidPosts = [];
+        adList.forEach((post) => {
+            if (AdCollection._validate(post) && this.get(post.id)) {
+                this._adList.push(post);
+            } else {
+                notValidPosts.push(post);
+            }
+        })
+        return notValidPosts;
+    }
+
+    getPage(skip, top, filterConfig) {
+        let countSkippedPosts = skip || 0;
+        let countReceivedPosts = top || 10;
+        let workingArray = [...this._adList];
+        if (filterConfig) {
+            if (filterConfig.author) {
+                workingArray = workingArray.filter(value => value.author === filterConfig.author)
+            }
+            if (filterConfig.createdAt) {
+                workingArray = workingArray.filter(value => {
+                    return value.createdAt.getDate() === filterConfig.createdAt.getDate()
+                        && value.createdAt.getMonth() === filterConfig.createdAt.getMonth()
+                        && value.createdAt.getFullYear() === filterConfig.createdAt.getFullYear();
+                })
+            }
+            if (filterConfig.tags) {
+                workingArray = workingArray.filter(value => {
+                    let flag = false;
+                    value.hashTags.forEach(tag => {
+                            if (filterConfig.tags.includes(tag)) {
+                                flag = true;
+                            }
+                        }
+                    )
+                    return flag;
+                })
+            }
+            if (filterConfig.validateUntil) {
+                workingArray = workingArray.filter(value => value.validateUntil - filterConfig.validateUntil >= 0)
+            }
+            if (filterConfig.discount) {
+                workingArray = workingArray.filter(value => value.discount - filterConfig.discount >= 0)
```

Comment:
Why not just `value.discount >= filterConfig.discount`? The same for lines 57 and 64